### ISSUE TITLE
As a user I can rank patients by their scheduled time

### ISSUE THREAD

**Issue by shenchenzizoe** [AY2223S2-CS2103T-W11-4/tp#49]

**Comment by shenchenzizoe** [comment]
Added a new 'time' field to Person to represent the scheduled time, it's alternative so can be null. A new TimeComparator Class to sort Persons by the scheduled time.

**Comment by shenchenzizoe** [comment]
Change in PersonListCard.fxml and PersonCard profile to add a label 'time' that shows Person's scheduled time.
The constructor of JsonAdaptedPerson Class, however, is not modified accordingly in correct way, waiting for future improvements.
Add a command ListTime to sort Persons with scheduled time in order. In UI the input should be 'listTime' but the feature is not implemented correctly, the filteredPersons is of the class FilteredList, which can not be sorted. Thus, the updateScheduledTime method in class ModelManager fails. Future modifications are needed to show the filtered and sorted list in the MainWindow.

**Comment by shenchenzizoe** [comment]
Resolved the bug of sorting the list !!!!
However, commented on three test cases in EditCommand and DeleteCommand, 
because I have modified the filteredPersons in ModelManager.
Initially, data in the address book is directly passed into the filteredPersons. Now it's coped into ObservableList persons and then passed to filteredPersons, so the persons can be reassigned with the list we have sorted, which will be shown in UI. 
The updateFilteredPersonList(Predicate<Person> predicate) method is also modified so that the ObservableList persons can grab data from the address book again and we get our complete list in UI.
